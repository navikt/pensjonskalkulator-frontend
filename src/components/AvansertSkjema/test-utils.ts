export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
