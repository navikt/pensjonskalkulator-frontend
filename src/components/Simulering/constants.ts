export const MAX_UTTAKSALDER = 77
export const COLUMN_WIDTH = 25
export const TOOLTIP_YPOS = 35

export const SERIES_COLORS = {
  SERIE_COLOR_INNTEKT: 'var(--a-gray-500)',
  SERIE_COLOR_AFP: 'var(--a-purple-400)',
  SERIE_COLOR_TP: 'var(--a-green-400)',
  SERIE_COLOR_ALDERSPENSJON: 'var(--a-deepblue-500)',
  SERIE_COLOR_FADED_INNTEKT: 'var(--a-gray-300)',
  SERIE_COLOR_FADED_AFP: 'var(--a-purple-200)',
  SERIE_COLOR_FADED_TP: 'var(--a-green-200)',
  SERIE_COLOR_FADED_ALDERSPENSJON: 'var(--a-deepblue-200)',
}

export const SERIES_DEFAULT = {
  SERIE_INNTEKT: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'Pensjonsgivende inntekt',
    color: SERIES_COLORS.SERIE_COLOR_INNTEKT,
  },
  SERIE_TP: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'Pensjonsavtaler (arbeidsgivere m.m.)',
    color: SERIES_COLORS.SERIE_COLOR_TP,
  },
  SERIE_AFP: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'AFP (Avtalefestet pensjon)',
    color: SERIES_COLORS.SERIE_COLOR_AFP,
  },
  SERIE_ALDERSPENSJON: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'Alderspensjon (NAV)',
    color: SERIES_COLORS.SERIE_COLOR_ALDERSPENSJON,
  },
}

export const highchartsScrollingSelector = '.highcharts-scrolling'
