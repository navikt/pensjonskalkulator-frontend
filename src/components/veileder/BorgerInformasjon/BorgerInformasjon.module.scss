@use '../../../scss/variables';

.wrapper {
  padding: var(--a-spacing-4) var(--a-spacing-6);
  font-weight: bold;
}
