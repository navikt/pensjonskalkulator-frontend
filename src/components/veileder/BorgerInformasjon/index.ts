export { BorgerInformasjon as default } from './BorgerInformasjon'
