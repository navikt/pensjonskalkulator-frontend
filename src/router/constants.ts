export const BASE_PATH = '/pensjon/kalkulator'

export const externalUrls = {
  dinPensjon: 'https://nav.no/pensjon',
  dinPensjonBeholdning: 'https://www.nav.no/pensjon/opptjening/nb/',
  detaljertKalkulator: 'https://www.nav.no/pselv/simulering.jsf',
  alderspensjonsregler: 'https://www.nav.no/alderspensjon#beregning',
  afp: 'https://www.afp.no',
  afpPrivat: 'https://www.nav.no/afp-i-privat-sektor',
  garantipensjon: 'https://www.nav.no/minstepensjon',
  norskPensjon: 'https://norskpensjon.no/',
  uinnloggetKalkulator:
    'https://www.nav.no/pselv/simulering/forenkletsimulering.jsf',
  personvernerklaering:
    'https://www.nav.no/personvernerklaering#dine-rettigheter',
  personvernerklaeringKontaktOss:
    'https://www.nav.no/personvernerklaering#kontakt-nav',
  kontaktOss: 'https://www.nav.no/kontaktoss',
  planleggePensjon:
    'https://www.nav.no/planlegger-pensjon#noe-du-ikke-finner-svaret-p-her',
}

export const paths = {
  root: '/',
  login: '/login',
  henvisning: '/henvisning',
  start: '/start',
  samtykke: '/samtykke',
  utenlandsopphold: '/utenlandsopphold',
  offentligTp: '/offentlig-tp',
  afp: '/afp',
  samtykkeOffentligAFP: '/samtykke-offentlig-afp',
  ufoeretrygdAFP: '/ufoeretrygd-afp',
  sivilstand: '/sivilstand',
  uventetFeil: '/uventet-feil',
  beregningEnkel: '/beregning',
  beregningAvansert: '/beregning-detaljert',
  forbehold: '/forbehold',
  personopplysninger: '/personopplysninger',
} as const

export const henvisningUrlParams = {
  foedselsdato: '1963',
  apotekerne: 'apotekerne',
  utland: 'utland',
} as const
