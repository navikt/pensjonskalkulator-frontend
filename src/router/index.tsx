export { routes } from './routes'
export { externalUrls, paths, BASE_PATH } from './constants'
