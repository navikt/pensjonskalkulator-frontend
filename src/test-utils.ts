import renderer from 'react-test-renderer'

import { render } from '@testing-library/react'

export function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
