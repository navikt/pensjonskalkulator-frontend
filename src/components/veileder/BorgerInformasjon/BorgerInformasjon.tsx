import {
  Box,
  HStack,
  Loader,
  CopyButton,
  Spacer,
  Button,
  BodyShort,
  BodyLong,
} from '@navikt/ds-react'

import { BASE_PATH } from '@/router/constants'
import { useGetPersonQuery } from '@/state/api/apiSlice'

import styles from './BorgerInformasjon.module.scss'

interface IBorgerInformasjonProps {
  fnr: string
  children?: React.ReactNode
}
const formatFnr = (fnr: string) => {
  return `${fnr.slice(0, 6)} ${fnr.slice(6)}`
}

export const BorgerInformasjon: React.FC<IBorgerInformasjonProps> = ({
  fnr,
  children,
}) => {
  const { data: person, isFetching: isPersonFetching } = useGetPersonQuery()

  const onNullstillClick = () => {
    window.location.href = `${BASE_PATH}/veileder`
  }

  return (
    <Box
      background="bg-default"
      borderWidth="0 0 1 0"
      borderColor="border-divider"
    >
      <HStack align="center" gap="2" className={styles.wrapper} style={{}}>
        <BodyLong weight="semibold">
          {isPersonFetching ? <Loader /> : person?.navn}
        </BodyLong>
        <span aria-hidden="true">/</span>
        <HStack align="center" gap="1">
          <BodyShort data-testid="borger-fnr" size="small" weight="semibold">
            F.nr.: {formatFnr(fnr)}
          </BodyShort>
          <CopyButton size="small" copyText={fnr} />
        </HStack>
        <Spacer />
        <HStack gap="2">
          {children}
          <Button
            onClick={onNullstillClick}
            data-testid="borger-nullstill"
            variant="secondary"
            size="small"
          >
            Nullstill bruker
          </Button>
        </HStack>
      </HStack>
    </Box>
  )
}
