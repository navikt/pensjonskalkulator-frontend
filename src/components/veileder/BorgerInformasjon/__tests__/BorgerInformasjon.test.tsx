import { BASE_PATH } from '@/router/constants'
import { render, screen, userEvent, waitFor } from '@/test-utils'

import { BorgerInformasjon } from '../BorgerInformasjon'

const previousWindow = window

describe('veileder - BorgerInformasjon', async () => {
  afterEach(() => {
    global.window = previousWindow
  })

  it('viser informasjon og fnr', async () => {
    render(<BorgerInformasjon fnr="12345678901" />)

    await waitFor(() => {
      expect(screen.getByTestId('borger-fnr')).toHaveTextContent(
        'F.nr.: 123456 78901'
      )
    })
  })

  it('reloader siden om man trykker pÃ¥ nullstill', async () => {
    const user = userEvent.setup()
    render(<BorgerInformasjon fnr="12345678901" />)
    global.window = Object.create(window)
    Object.defineProperty(window, 'location', {
      value: {
        href: 'before',
      },
      writable: true,
    })

    await waitFor(async () => {
      expect(screen.getByTestId('borger-nullstill')).toBeInTheDocument()
      expect(window.location.href).toBe('before')
      await user.click(screen.getByTestId('borger-nullstill'))
      expect(window.location.href).toBe(`${BASE_PATH}/veileder`)
    })
  })
})
