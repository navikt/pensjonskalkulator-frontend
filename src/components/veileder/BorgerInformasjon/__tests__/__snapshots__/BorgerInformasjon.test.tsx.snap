// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`veileder - BorgerInformasjon > viser informasjon og fnr 1`] = `
<DocumentFragment>
  <div
    class="navds-box navds-box-bg navds-box-border-color navds-box-border-width"
    style="--__ac-box-background: var(--a-bg-default); --__ac-box-border-color: var(--a-border-divider); --__ac-box-border-width: 0px 0px 2px 0px;"
  >
    <div
      class="navds-stack _wrapper_f04f26 navds-hstack"
      style="--__ac-stack-wrap: wrap; --__ac-stack-gap-xs: var(--a-spacing-2); --__ac-stack-direction-xs: row; --__ac-stack-align-xs: center;"
    >
      <svg
        aria-labelledby="loader-r0"
        class="navds-loader navds-loader--medium navds-loader--neutral"
        focusable="false"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 50 50"
      >
        <title
          id="loader-r0"
        >
          venter...
        </title>
        <circle
          class="navds-loader__background"
          cx="25"
          cy="25"
          fill="none"
          r="20"
          xmlns="http://www.w3.org/2000/svg"
        />
        <circle
          class="navds-loader__foreground"
          cx="25"
          cy="25"
          fill="none"
          r="20"
          stroke-dasharray="50 155"
        />
      </svg>
      <span
        aria-hidden="true"
      >
        /
      </span>
      <div
        class="navds-stack navds-hstack"
        style="--__ac-stack-wrap: wrap; --__ac-stack-gap-xs: var(--a-spacing-1); --__ac-stack-direction-xs: row; --__ac-stack-align-xs: center;"
      >
        <div
          data-testid="borger-fnr"
        >
          F.nr.: 12345678901
        </div>
        <button
          aria-live="polite"
          class="navds-copybutton navds-copybutton--small navds-copybutton--neutral navds-copybutton--icon-only"
          type="button"
        >
          <span
            class="navds-copybutton__content"
          >
            <span
              class="navds-copybutton__icon"
            >
              <svg
                aria-hidden="false"
                aria-labelledby="title-r1"
                fill="none"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="title-r1"
                >
                  Kopier
                </title>
                <path
                  clip-rule="evenodd"
                  d="M8.25 3.5c0-.69.56-1.25 1.25-1.25H14a.75.75 0 0 1 .53.22l5 5c.141.14.22.331.22.53v8.5c0 .69-.56 1.25-1.25 1.25h-9c-.69 0-1.25-.56-1.25-1.25zm6.25 5.25c-.69 0-1.25-.56-1.25-1.25V3.75h-3.5v12.5h8.5v-7.5zm.25-3.94 2.44 2.44h-2.44zM6.502 7.75H5.75v12.5h8.5v-.748a.75.75 0 0 1 1.5 0v.998c0 .69-.56 1.25-1.25 1.25h-9c-.69 0-1.25-.56-1.25-1.25v-13c0-.69.56-1.25 1.25-1.25h1.002a.75.75 0 1 1 0 1.5"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
      <span
        class="navds-stack__spacer"
      />
      <div>
        <button
          class="navds-button navds-button--secondary navds-button--small"
          data-testid="borger-nullstill"
        >
          <span
            class="navds-label navds-label--small"
          >
            Nullstill bruker
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
